lxd/storage_volumes_snapshot.go

```diff
@@ -724,13 +700,85 @@ func autoCreateCustomVolumeSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
 				continue
 			}
 
-			volumes = append(volumes, v)
+			// Always include local volumes
+			if v.NodeID == nodeID {
+				volumes = append(volumes, v)
+			} else if v.NodeID < 0 {
+				// Keep a separate list of remote volumes
+				remoteVolumes = append(remoteVolumes, v)
+			}
 		}
 
-		if len(volumes) == 0 {
+		// Exit early if there are no volumes to be snapshotted.
+		if len(volumes) == 0 && len(remoteVolumes) == 0 {
 			return
 		}
 
+		var nodes []api.ClusterMember
+		var availableNodeIDs []int64
+
+		if len(remoteVolumes) > 0 {
+			isClustered, err := cluster.Enabled(d.State().Node)
+			if err != nil {
+				return
+			}
+
+			if isClustered {
+				// Get list of cluster nodes
+				nodes, err = cluster.List(d.State(), d.gateway)
+				if err != nil {
+					return
+				}
+
+				for _, node := range nodes {
+					var nodeID int64
+
+					if node.Status == "Online" {
+						err := d.cluster.Transaction(func(tx *db.ClusterTx) error {
+							info, err := tx.GetNodeByName(node.ServerName)
+							if err != nil {
+								return err
+							}
+
+							nodeID = info.ID
+							return nil
+						})
+						if err != nil {
+							logger.Warnf("%s", errors.Wrap(err, "Skipping auto custom volume snapshots"))
+							return
+						}
+
+						availableNodeIDs = append(availableNodeIDs, nodeID)
+					}
+				}
+			}
+
+			if isClustered && len(availableNodeIDs) <= 0 {
+				logger.Warn("Skipping auto custom volume snapshots")
```

Comment:
@monstermunchkin could you add some info as to why its being skipped, something like ", due to no online cluster members"